src/python/grpcio/grpc/_channel.py

```diff
@@ -728,7 +731,10 @@ def stop_channel_spin(timeout):  # pylint: disable=unused-argument
                     call.cancel()
 
     channel_spin_thread = _common.CleanupThread(
-        stop_channel_spin, target=channel_spin)
+        stop_channel_spin,
+        target=channel_spin,
+        name='Thread-gRPC-StopChannelSpin',
```

Comment:
@nathanielmanistaatgoogle This should probably be named `Thread-gRPC-ChannelSpin`, when I wrote it I didn't realize that the `target` was more relevant than the cleanup (this thread doesn't even get returned, so for someone to call `.join()` on it, they'd have to grab it from `threading.enumerate()` or similar).

FWIW I'm not entirely sure which channel is being spun or why.

Reply:
As for the name: yes, I think `channel_spin` being the target is controlling and the thread should be named `Thread-gRPC-channel_spin`.

As for why the thread exists: this is the thread that drives gRPC Core when the application, having made an asynchronous call, hasn't donated one of its own threads for the purpose.

Reply:
Thanks for the "why exists?" answer